pkg/docker/config/config.go

```diff
@@ -55,7 +56,27 @@ var (
 
 // SetAuthentication stores the username and password in the auth.json file
 func SetAuthentication(sys *types.SystemContext, registry, username, password string) error {
+	credConfigs, err := sysregistriesv2.GetRegistryCredentialHelpers(sys, registry)
+	if err != nil {
+		return err
+	}
+	if len(credConfigs) > 0 {
+		if err := func(credConfigs []string, registry, username, password string) error {
```

Comment:
I don’t understand why the `func() ()` parts exist. Wouldn’t it be simpler to inline fully? Even the `len(credConfigs) > 0` check would go away.